modules/nlu/src/backend/pipelines/intents/predictions_utils.ts

```diff
@@ -0,0 +1,112 @@
+import * as sdk from 'botpress/sdk'
+import _ from 'lodash'
+import math from 'mathjs'
+
+import { GetZPercent } from '../../tools/math'
+import { LanguageProvider, Token2Vec } from '../../typings'
+import { getSentenceFeatures } from '../language/ft_featurizer'
+
+// this means that the 3 best predictions are really close, do not change magic numbers
+const predictionsReallyConfused = (predictions: sdk.MLToolkit.SVM.Prediction[]) => {
+  const intentsPreds = predictions.filter(x => x.label !== 'none')
+  if (intentsPreds.length <= 2) {
+    return false
+  }
+
+  const std = math.std(intentsPreds.map(p => p.confidence))
+  const diff = (intentsPreds[0].confidence - intentsPreds[1].confidence) / std
+
+  if (diff >= 2.5) {
+    return false
+  }
+
+  const bestOf3STD = math.std(predictions.slice(0, 3).map(p => p.confidence))
+  return bestOf3STD <= 0.03
+}
+
+const predictL0Contextually = async function(
+  l0Features: number[],
+  includedContexts: string[],
+  l0Predictor
+): Promise<sdk.MLToolkit.SVM.Prediction[]> {
+  const allL0 = await l0Predictor.predict(l0Features)
+  const includedL0 = allL0.filter(c => includedContexts.includes(c.label))
+  const totalL0Confidence = Math.min(1, _.sumBy(includedL0, c => c['confidence']))
```

Comment:
can be just a string (`_.sumBy(l0, 'confidence')`) but ultimately we want the typing to catch refactoring errors

Reply:
Not sure what you mean here? Can you explain more :)

Reply:
lodash iteratee params don't have to be functions, you can pas a string directly and it'll do the same